src/Filesystem/Filesystem.php

```diff
@@ -120,49 +120,57 @@ public function removeFile(Media $media, string $path)
         $this->filesystem->disk($media->disk)->delete($path);
     }
 
-    public function syncFileNames(Media $media)
+    public function syncFiles(Media $media)
     {
-        $this->renameMediaFile($media);
+        $this->syncMediaFile($media);
 
-        $this->renameConversionFiles($media);
+        $this->syncConversionFiles($media);
     }
 
-    protected function renameMediaFile(Media $media)
+    protected function syncMediaFile(Media $media)
     {
         $newFileName = $media->file_name;
         $oldFileName = $media->getOriginal('file_name');
 
+        $newDisk = $media->disk;
+        $oldDisk = $media->getOriginal('disk');
+
         $mediaDirectory = $this->getMediaDirectory($media);
+        $oldMediaDirectory = $this->getMediaDirectory($media->setAttribute('disk', $oldDisk));
+        $media->setAttribute('disk', $newDisk);
 
-        $oldFile = $mediaDirectory.'/'.$oldFileName;
-        $newFile = $mediaDirectory.'/'.$newFileName;
+        $oldFile = $oldMediaDirectory.$oldFileName;
+        $newFile = $mediaDirectory.$newFileName;
 
-        $this->filesystem->disk($media->disk)->move($oldFile, $newFile);
+        $this->move($oldDisk, $oldFile, $newDisk, $newFile);
     }
 
-    protected function renameConversionFiles(Media $media)
+    protected function syncConversionFiles(Media $media)
     {
         $newFileName = $media->file_name;
         $oldFileName = $media->getOriginal('file_name');
 
+        $newDisk = $media->disk;
+        $oldDisk = $media->getOriginal('disk');
+
         $conversionDirectory = $this->getConversionDirectory($media);
+        $oldConversionDirectory = $this->getConversionDirectory($media->setAttribute('disk', $oldDisk));
+        $media->setAttribute('disk', $newDisk);
 
         $conversionCollection = ConversionCollection::createForMedia($media);
```

Comment:
Wouldn't it be easier to just move the entire contents of the conversions directory (and the responsive images directory)?